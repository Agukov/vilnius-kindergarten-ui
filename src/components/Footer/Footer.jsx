import React from 'react'
import propTypes from 'prop-types'
import {Panel} from 'react-bootstrap'

export class Footer extends React.Component {
  render(){
      return (
        <Panel>
          Fancy footer <br/>
          contacts...
        </Panel>
      )
  }
}
