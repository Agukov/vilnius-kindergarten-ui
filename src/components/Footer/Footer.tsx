import * as React from 'react';
import { Container, Row, Col, Alert } from 'reactstrap';
import { InlineForm } from '../InlineForm/InlineForm';
import './Footer.css';

// define state interface
export interface FooterState {
  nameValue: string;
  emailValue: string;
  isSuccesMessage: boolean;
}

export class Footer extends React.Component<{}, FooterState> {
  // in constructor we define initial state
  constructor(props: FooterState) {
    super(props);
    this.state = {
      nameValue: '',
      emailValue: '',
      isSuccesMessage: false
    };
  }

  // call this function whent you want to render succes method
  renderSuccesMessage = (): JSX.Element => {
    return(
      <Alert color="info">
        <strong>You have just registered to newsletter</strong>
        <div>
          name: <b>{this.state.nameValue}</b><br/>
          email: <b>{this.state.emailValue}</b><br/>
        </div>
      </Alert>
    );
  }
  // pass this function to InlineForm props to move data up
  onHandleContactsFormSubmit = (event: React.FormEvent<HTMLFormElement>): void => {
    // prevents default action when form is being submitted. In this case we prevent refreshing page
    event.preventDefault();
    // expr1 && expr2 - if expr1 == true, then execute expr2

    if (this.state.emailValue && this.state.nameValue) {
      this.setState({isSuccesMessage : true});
    }
  }
  // pass this function to InlineForm props to move data up
  onHandleInputNameChange = (event: React.FormEvent<HTMLInputElement>): void => {
    this.setState({nameValue: event.currentTarget.value});
  }
  // pass this function to InlineForm props to move data up
  onHandleInputEmailChange = (event: React.FormEvent<HTMLInputElement>): void => {
    this.setState({emailValue: event.currentTarget.value});
  }

  render() {
    return(
      <div className="app-footer">
        <Container>
         <Row>
           <Col xs="3">some cool links</Col>
           <Col xs="6">
            {this.state.isSuccesMessage && this.renderSuccesMessage()}
           </Col>
           <Col xs="3">
             <InlineForm
                onHandleContactsFormSubmit={this.onHandleContactsFormSubmit}
                onHandleInputNameChange={this.onHandleInputNameChange}
                onHandleInputEmailChange={this.onHandleInputEmailChange}
             />
           </Col>
         </Row>
       </Container>
      </div>
    );
  }
}
