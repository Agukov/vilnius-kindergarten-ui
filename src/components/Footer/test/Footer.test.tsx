import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Footer } from '../Footer';

describe('Footer Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Footer/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctle(checked with enzyme)', () => {
    // console.log(shallow(<Footer/>).debug());
    const hasClass: boolean = shallow(<Footer/>).hasClass('app-footer');
    expect(hasClass).toBeTruthy();
  });
});

it('renders mock date', () => {
  Date.now = jest.fn(() => 14823633672);
  expect(Date.now()).toMatchSnapshot();
});
