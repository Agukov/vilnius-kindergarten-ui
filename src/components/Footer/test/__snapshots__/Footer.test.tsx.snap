// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders correctly 1`] = `
<div
  className="app-footer"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-3"
      >
        somea cool links
      </div>
      <div
        className="col-6"
      />
      <div
        className="col-3"
      >
        <form
          onSubmit={[Function]}
        >
          <div>
            <input
              className="form-control"
              name="name"
              onChange={[Function]}
              placeholder="name"
              type="text"
            />
            <input
              className="form-control"
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="email"
            />
            <button
              className="btn btn-secondary"
              onClick={[Function]}
              type={undefined}
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders mock date 1`] = `14823633672`;
